hail/src/main/scala/is/hail/utils/Interval.scala

```diff
@@ -119,14 +123,30 @@ class Interval(val left: IntervalEndpoint, val right: IntervalEndpoint) extends
 
   def hull(pord: ExtendedOrdering, other: Interval): Interval =
     Interval(
-      ext(pord).min(this.left, other.left).asInstanceOf[IntervalEndpoint],
-      ext(pord).max(this.right, other.right).asInstanceOf[IntervalEndpoint])
+      // min(this.left, other.left)
+      if (ext(pord).lt(this.left, other.left))
```

Comment:
I feel like these should maybe be using `compare`, no? otherwise (I think) we can get things like 
```
hull([-0.0, 1.0], [0.0, 1.0]) = [0.0, 1.0]
hull([0.0, 1.0], [-0.0, 1.0]) = [-0.0, 1.0]
```

Reply:
Yeah.  In fact, I think Interval should use compare (total ordering) everywhere.  That matches the old behavior, too.  Fixed.